### DISCUSSION TITLE
extractSignatures error

### DISCUSSION THREAD

**Discussion by pradeep-sidd** [PoisonAlien/maftools#945] [General]
HI everyone, 
I am performing mutational signature analysis on a large dataset of esophageal cancer samples, however when I use the estimatesignatures function using te created tnm list i get this error notification - "Error in cons != .consold : non-conformable arrays". I removed all the NAs from the list but the error is persisting. I am attaching a screenshot of the code and the error message.
I am using the maftools package. I am stuck and this is essential to my proceed further. Really appreciate your help with this.

> combined.sig = extractSignatures(mat = eso.tnm, n = 2)
-Running NMF for factorization rank: 2
Error in cons != .consold : non-conformable arrays
Timing stopped at: 0.3 0 0.4
> young.sig = extractSignatures(mat = young.tnm, n =3)
-Found zero mutations for conversions:
  C[T>G]A
  T[T>G]G
-Running NMF for factorization rank: 3
Error: NMF::nmf - Input matrix x contains at least one null or NA-filled row.
> old.sig = extractSignatures(mat = old.tnm, n =3 )
-Running NMF for factorization rank: 3
Error in cons != .consold : non-conformable arrays

**Comment by ShixiangWang** [comment]
add a very small constant to the input matrix with option `pConstant`, like 1e-6, and take a try?

**Comment by pradeep-sidd** [comment]
Thank you, it worked.

Pradeep

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
